### PR TITLE
Add assessor data to redux store

### PR DESCRIPTION
**Pull request by adi130987**
* add assessor to redux store
* fix unnecessary assessor req, resulting in 404 when assessment has no assessor saved yet

### CODE REVIEW

File: src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts [hunk 1]
```diff
@@ -186,7 +186,8 @@ export class AssessorInfo extends connect(store)(LitElement) {
       this.assessor = new Assessor();
       return;
     }
-    const url = getEndpoint(etoolsEndpoints.assessor, {id: assessmentId}).url!;
+    if (this.assessor !== undefined) {
```
Comment (acory) [inline src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts]: ???

Comment (adi130987) [reply src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts]: :) updated

File: src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts [hunk 2]
```diff
@@ -172,46 +171,35 @@ export class AssessorInfo extends connect(store)(LitElement) {
...
-      const assessmentId = state.app!.routeDetails.params.assessmentId;
-      if (this.assessmentId !== assessmentId) {
-        this.assessmentId = assessmentId;
-        this.getAssessorDetails(this.assessmentId);
+
+    // initialize assessment object from redux state
+    if (state.pageData!.currentAssessment) {
+      const newAssessment = state.pageData!.currentAssessment;
+      if (!isJsonStrMatch(this.assessment, newAssessment)) {
+        this.assessment = cloneDeep(newAssessment);
```
Comment (acory) [inline src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts]: Seems that only assessmentId is needed in this component

File: src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts [hunk 3]
```diff
@@ -267,40 +254,18 @@ export class AssessorInfo extends connect(store)(LitElement) {
     this.requestUpdate();
   }
 
-  saveAssessor() {
+  saveAssessorChanges() {
```
Comment (acory) [inline src_ts/components/pages/assessments/assessment-tab-pages/details/assessor-info.ts]: Rename doesn't seem necessary

File: src_ts/redux/actions/page-data.ts [hunk 4]
```diff
@@ -18,6 +25,60 @@ export const updateAssessmentData: ActionCreator<AssessmentActionUpdate> = (curr
...
+      logError(err);
+      if (err.status === 404) {
+        // in case assessor is not found, init as new assessor
+        dispatch(updateAssessorData(new Assessor()));
+      }
+    });
+};
+
+/**
+ * Save assessment assessor data
```
Comment (acory) [inline src_ts/redux/actions/page-data.ts]: When the comment is the same as the method name is not needed

**Pull request merged**